### PR TITLE
fix: hotfix v6.89.1 for proper error handling in encrypt-submission middleware

### PR DESCRIPTION
**Pull request by wanlingt**
## Problem

Docker containers were crashing upon because of this error `Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client`

This was due to incorrect error handling - `res.status.json` was being set twice

## Solution

Set and send res.status.json only once

### CODE REVIEW

**Review by tshuli — APPROVED** [review]
lgtm! L433 to 456 can be taken out entirely:

1) In L455, we already return error. so whether or not it is hard validation, we are returning error response. So the log line that 'Error processing responses, but proceeding with submission as submission have been validated client-side', is no longer valid
2) But why do we return error - i think this was actually a mistake in the PR 😁 https://github.com/opengovsg/FormSG/pull/6898/files#diff-f6575f85355753bb9d49c6d9ff5953bd0f4cff0c5b138091f2c28b64c25149beR457

The original logic (altered by the PR) was that if there is no hard validation, then you should return next() if there is validation error.

TLDR - remove L433 to 456 and we should be good to go 😃
